engine/common/synchronization/engine.go

```diff
@@ -509,8 +509,72 @@ func (e *Engine) sendRequests(participants flow.IdentifierList, ranges []chainsy
+
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to generate random number from %x: %w", originID[:], err)
+	}
+
+	// validity check: if no block IDs, always report as misbehavior
+	if len(batchRequest.BlockIDs) == 0 {
+		e.log.Warn().
+			Hex("origin_id", logging.ID(originID)).
+			Str(logging.KeySuspicious, "true").
+			Str("reason", alsp.InvalidMessage.String()).
+			Msg("received invalid batch request with 0 block IDs, creating ALSP report")
+		report, err := alsp.NewMisbehaviorReport(originID, alsp.InvalidMessage)
+
+		if err != nil {
+			// failing to create the misbehavior report is unlikely. If an error is encountered while
+			// creating the misbehavior report it indicates a bug and processing can not proceed.
+			return nil, false, fmt.Errorf("failed to create misbehavior report (invalid batch request, no block IDs) from %x: %w", originID[:], err)
+		}
+		// failed validation check and should be reported as misbehavior
+		return report, false, nil
+	}
+
+	// to avoid creating a misbehavior report for every batch request received, use a probabilistic approach.
+	// The larger the batch request and base probability, the higher the probability of creating a misbehavior report.
+
+	// batchRequestProb is calculated as follows:
+	// batchRequestBaseProb * (len(batchRequest.BlockIDs) + 1) / synccore.DefaultConfig().MaxSize
+	// Example 1 (small batch of block IDs) if the batch request is for 10 blocks IDs and batchRequestBaseProb is 0.01, then the probability of
+	// creating a misbehavior report is:
+	// batchRequestBaseProb * (10+1) / synccore.DefaultConfig().MaxSize
+	// = 0.01 * 11 / 64 = 0.00171875 = 0.171875%
+	// Example 2 (large batch of block IDs) if the batch request is for 1000 block IDs and batchRequestBaseProb is 0.01, then the probability of
+	// creating a misbehavior report is:
+	// batchRequestBaseProb * (1000+1) / synccore.DefaultConfig().MaxSize
+	// = 0.01 * 1001 / 64 = 0.15640625 = 15.640625%
+	batchRequestProb := e.spamDetectionConfig.batchRequestBaseProb * (float32(len(batchRequest.BlockIDs)) + 1) / float32(synccore.DefaultConfig().MaxSize)
+	if float32(n) < batchRequestProb*spamProbabilityMultiplier {
+		// create a misbehavior report
+		e.log.Warn().
+			Hex("origin_id", logging.ID(originID)).
+			Str(logging.KeySuspicious, "true").
+			Str("reason", alsp.ResourceIntensiveRequest.String()).
+			Msgf("for %d block IDs, creating probabilistic ALSP report", len(batchRequest.BlockIDs))
+		report, err := alsp.NewMisbehaviorReport(originID, alsp.ResourceIntensiveRequest)
+
+		if err != nil {
+			// failing to create the misbehavior report is unlikely. If an error is encountered while
+			// creating the misbehavior report it indicates a bug and processing can not proceed.
+			return nil, false, fmt.Errorf("failed to create misbehavior report from %x: %w", originID[:], err)
+		}
+		// failed validation check and should be reported as misbehavior
+		return report, false, nil
```

Comment:
If we return `valid=false` here, the caller will both create a misbehaviour report and discard the message. 

The approach we are taking here to report ambiguous misbehaviour (essentially: load) is to probabilistically create MisbehaviourReports based on message size. Sometimes we will report innocuous messages. On average, we will create more reports for senders which send us more and larger requests.

However, since we are measuring load (not unambiguous misbehaviour), I don't think it makes sense to discard these messages. In practice, we are just introducing another source of random message loss. In the happy path, there are no misbehaving nodes, and we will randomly discard about 1% of all messages.

Reply:
As a side note, when we create this class of misbehaviour report, we will also [log at warn level with the `Suspicious` key](https://github.com/onflow/flow-go/blob/9b4f52947d8efefb6904041f11a1fcabcdeb5b64/engine/common/synchronization/engine.go#L217-L221). This will randomly add spurious warnings to our logs for what are really innocuous, expected messages. 

Another reason to conceptually differentiate between unambiguous misbehaviour and measuring load.

Reply:
[TODO] update to remove boolean and not drop any messages. Retrofit to use for SyncRequest and RangeRequest.

Reply:
Captured requirement to not log load related misbehavior reports with `Suspicious` key in https://github.com/dapperlabs/flow-go/issues/6880 for future PR since there's no distinction currently between ambiguous and unambiguous misbehavior.

Reply:
updated validation functions:
- only return (irrecoverable) error and not return misbehavior report nor boolean
- reports misbehavior as soon as its found instead of returning to caller
- classify ambiguous misbehavior reports with new key (`KeyLoad`) to differentiate misbehavior reports that are unambiguously malicious / wrong (`KeySuspicious`)